core/tests/environment.ts

```diff
@@ -0,0 +1,52 @@
+import { createEnv } from '@t3-oss/env-core';
+import { config as dotenvConfig } from 'dotenv';
+import { z } from 'zod/v4';
+
+import { defaultLocale, locales } from '~/i18n/locales';
+
+dotenvConfig({ path: ['.env', '.env.local', '.env.test'], override: true });
+
+const types = {
+  string: z.string().optional(),
+  bool: z.stringbool().optional(),
+  number: z.string().transform(Number).optional(),
```

Comment:
I think the `.coerce` utility is better here instead of `.transform`
```suggestion
  number: z.coerce.number().optional(),
```

Reply:
@chanceaclark Ahh good call, I was using `z.number({ coerce: true })` previously, but it wasn't available anymore after upgrading so I thought it was removed. I didn't realize `z.coerce` was a thing still 👍